src/Microsoft.DotNet.ProjectModel.Server/ApplicationContext.cs

```diff
@@ -0,0 +1,455 @@
+
+            _log.LogInformation("Finish draining inbox.");
+        }
+
+        private bool ProcessMessage()
+        {
+            Message message;
+
+            lock (_inbox)
+            {
+                if (!_inbox.Any())
+                {
+                    return false;
+                }
+
+                message = _inbox.Dequeue();
+                Debug.Assert(message != null);
+            }
+
+            _log.LogInformation($"Received {message.MessageType}");
+
+            switch (message.MessageType)
+            {
+                case MessageTypes.Initialize:
+                    Initialize(message);
+                    break;
+                case MessageTypes.ChangeConfiguration:
+                    // TODO: what if the payload is null or represent empty string?
+                    _configure.Value = GetValue(message.Payload, "Configuration");
+                    break;
+                case MessageTypes.RefreshDependencies:
+                case MessageTypes.RestoreComplete:
+                    _refreshDependencies.Value = 0;
+                    break;
+                case MessageTypes.FilesChanged:
+                    _filesChanged.Value = 0;
+                    break;
+                case MessageTypes.GetDiagnostics:
+                    _waitingForDiagnostics.Add(message.Sender);
+                    break;
+            }
+
+            return true;
+        }
+
+        private void Initialize(Message message)
+        {
+            if (_initializedContext != null)
+            {
+                _log.LogWarning($"Received {message.MessageType} message more than once for {_appPath.Value}");
+                return;
+            }
+
+            _initializedContext = message.Sender;
+            _appPath.Value = GetValue(message.Payload, "ProjectFolder");
+            _configure.Value = GetValue(message.Payload, "Configuration") ?? "Debug";
+
+            var version = GetValue<int>(message.Payload, "Version");
+            if (version != 0 && !_protocolManager.EnvironmentOverridden)
+            {
+                _contextProtocolVersion = Math.Min(version, _protocolManager.MaxVersion);
+                _log.LogInformation($"Set context protocol version to {_contextProtocolVersion.Value}");
+            }
+        }
+
+        private bool ResolveDependencies()
+        {
+            ProjectState state = null;
+
+            if (_appPath.WasAssigned || _configure.WasAssigned || _filesChanged.WasAssigned || _refreshDependencies.WasAssigned)
+            {
+                bool triggerDependencies = _refreshDependencies.WasAssigned;
+
+                _appPath.ClearAssigned();
+                _configure.ClearAssigned();
+                _filesChanged.ClearAssigned();
+                _refreshDependencies.ClearAssigned();
+
+                state = _projectStateResolver.Resolve(_appPath.Value, _configure.Value, triggerDependencies, _remote.ProjectInformation?.ProjectSearchPaths);
```

Comment:
Nooo, can we not bring any of this stuff back and just simply use the workspace context?